users/dierkes/preprocessing/wav2vec.py

```diff
@@ -60,8 +79,155 @@ def run(self):
                 frames = soundfile.info(fname).frames
                 dest = train_f if rand.random() > self.valid_percent else valid_f
                 print(
-                    "{}\t{}".format(os.path.relpath(file_path, dir_path), frames), 
-                        file=dest
+                    "{}\t{}".format(os.path.relpath(file_path, dir_path), frames),
+                    file=dest,
                 )
         if valid_f is not None:
             valid_f.close()
+
+
+class FairseqWav2VecModelConvertAndDumpJob(Job):
+    """
+    Conversion Job for fairseq model checkpoints to RETURNN model dictionary and checkpoint
+    """
+
+    def __init__(
+        self,
+        wav2vec_config,
+        output_name,
+        audio_input,
+        device="cpu",
+        validate_atol=None,
+        fairseq_root=None,
+        pytorch_to_returnn_root=None,
+    ):
+        """
+        :param tk.Path wav2vec_config: path to wav2vec config to be converted
+        :param str output_name: name of resulting RETURNN model dictionary and checkpoint
+        :param tk.Path audio_input: path to exemplary audio file for forwarding through the model
+        :param str device: cpu or gpu
+        :param float|None validate_atol: allowed absolute difference between fairseq model outputs
+            and converted RETURNN outputs.
+        :param tk.Path|None fairseq_root: path to fairseq version to use for conversion
+        :param tk.Path|None pytorch_to_returnn_root: path to pytorch_to_returnn version to use for conversion
+        """
+        self.wav2vec_config = wav2vec_config
+        self.output_name = output_name
+        self.audio_input = audio_input
+        self.device = device
+        assert self.device in ["cpu", "gpu"]
+        self.validate_atol = validate_atol
+        self.fairseq_root = fairseq_root
+        self.pytorch_to_returnn_root = pytorch_to_returnn_root
+
+        if not self.fairseq_root:
+            print("WARNING: no explicit fairseq root directory given")
+        if not self.pytorch_to_returnn_root:
+            print("WARNING: no explicit pytorch_to_returnn root directory given")
+
+        self.returnn_model_out = self.output_path(self.output_name)
```

Comment:
```suggestion
        self.out_returnn_model = self.output_path(self.output_name)
```
see contributing guidelines

Reply:
Also, I would distinguish between the checkpoint and the config file with the network dict. Please create output variables for both of them. The checkpoint should be a `Checkpoint`, see `returnn/training.py`.